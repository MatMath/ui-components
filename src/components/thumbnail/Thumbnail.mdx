---
name: Thumbnail
menu: Components
---

import { Playground, Props } from 'docz';
import { Thumbnail } from './Thumbnail';
import '../../utility/playgroundStyles.css';

# Thumbnail

A thumbnail component

## Properties

<Props of={Thumbnail} />

## Usage

### Basic usage

Examples below show usage of the Thumbnail

<Playground
    style={{
        preview: {
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          justifyContent: 'space-evenly'
        }
    }}
>

  <Thumbnail size='small' text='Amazon' backgroundColor='grey' />
  <Thumbnail size='medium' text='Amazon' backgroundColor='grey' />
  <Thumbnail size='large' text='Amazon' backgroundColor='grey' />
  <Thumbnail
    size='large'
    text='Logo'
    backgroundColor='white'
    iconSource='https://cdn3.iconfinder.com/data/icons/glypho-social-and-other-logos/64/logo-amazon-512.png'
  />
  <Thumbnail
    size='medium'
    text='Logo'
    backgroundColor='orange'
    iconSource='https://cdn3.iconfinder.com/data/icons/glypho-social-and-other-logos/64/logo-amazon-512.png'
  />
  <Thumbnail
    size='small'
    text='Logo'
    backgroundColor='papayawhip'
    iconSource='https://cdn3.iconfinder.com/data/icons/glypho-social-and-other-logos/64/logo-amazon-512.png'
  />
</Playground>
