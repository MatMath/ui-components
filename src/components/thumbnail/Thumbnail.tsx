import * as React from 'react';
import { getClassNames } from '@utility/cssUtils';
import styles from './Thumbnail.module.scss';
import { white } from '@colors';

type ThumbnailSize = 'small' | 'medium' | 'large';

interface ThumbnailProps {
  size: ThumbnailSize;
  backgroundColor: string;
  text: string;
  iconSource?: string;
}

export const Thumbnail = ({
  size = 'medium',
  text = '',
  backgroundColor = white,
  iconSource
}: ThumbnailProps) => {
  return (
    <div
      className={getClassNames(styles.container, styles[size])}
      style={{ backgroundColor: backgroundColor }}
    >
      <Logo size={size} text={text} iconSource={iconSource} />
    </div>
  );
};

interface LogoProps {
  size: ThumbnailSize;
  iconSource?: string;
  text: string;
}

const Logo = ({ iconSource, text, size }: LogoProps) => {
  return (
    <>
      {iconSource ? (
        <span
          className={getClassNames(styles.icon, styles[size])}
          style={{ backgroundImage: `url(${iconSource})` }}
        />
      ) : (
        <span className={getClassNames(styles.letters, styles[size])}>
          {text.slice(0, 2)}
        </span>
      )}
    </>
  );
};
