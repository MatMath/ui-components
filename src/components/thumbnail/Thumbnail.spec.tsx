import * as React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import { Thumbnail } from './Thumbnail';

describe('<Thumbnail/>', () => {
  describe('Small render', () => {
    it('should render small', () => {
      const thumbnail = mount(
        <Thumbnail size='small' backgroundColor='gray' text='Logo' />
      );

      expect(thumbnail).toMatchSnapshot();
    });
  });

  describe('Medium render', () => {
    it('should render medium', () => {
      const thumbnail = mount(
        <Thumbnail size='medium' backgroundColor='gray' text='Logo' />
      );

      expect(thumbnail).toMatchSnapshot();
    });
  });

  describe('Large render', () => {
    it('should render large', () => {
      const thumbnail = mount(
        <Thumbnail size='large' backgroundColor='gray' text='Logo' />
      );

      expect(thumbnail).toMatchSnapshot();
    });
  });

  describe('Yellow render', () => {
    it('should render yello', () => {
      const thumbnail = mount(
        <Thumbnail size='small' backgroundColor='yellow' text='Logo' />
      );

      expect(thumbnail).toMatchSnapshot();
    });
  });

  describe('Text render', () => {
    it('should render text substring', () => {
      const thumbnail = mount(
        <Thumbnail
          size='small'
          backgroundColor='yellow'
          text='Some other text'
        />
      );

      const span = thumbnail.find('span');
      expect(span).toBeTruthy();
      expect(span.exists('.letters')).toEqual(true);
      expect(span.exists('.icon')).toEqual(false);
      expect(span.text()).toEqual('So');
    });
  });

  describe('Icon render', () => {
    it('should render an icon and no text', () => {
      const thumbnail = mount(
        <Thumbnail
          size='small'
          backgroundColor='yellow'
          text='Some other text'
          iconSource='fakelogo.png'
        />
      );

      const span = thumbnail.find('span');
      expect(span).toBeTruthy();
      expect(span.exists('.letters')).toEqual(false);
      expect(span.exists('.icon')).toEqual(true);
    });
  });
});
