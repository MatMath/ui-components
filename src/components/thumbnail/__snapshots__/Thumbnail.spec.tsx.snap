// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thumbnail/> Large render should render large 1`] = `
<Component
  backgroundColor="gray"
  size="large"
  text="Logo"
>
  <div
    className="container large"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  >
    <Logo
      size="large"
      text="Logo"
    >
      <span
        className="letters large"
      >
        Lo
      </span>
    </Logo>
  </div>
</Component>
`;

exports[`<Thumbnail/> Medium render should render medium 1`] = `
<Component
  backgroundColor="gray"
  size="medium"
  text="Logo"
>
  <div
    className="container medium"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  >
    <Logo
      size="medium"
      text="Logo"
    >
      <span
        className="letters medium"
      >
        Lo
      </span>
    </Logo>
  </div>
</Component>
`;

exports[`<Thumbnail/> Small render should render small 1`] = `
<Component
  backgroundColor="gray"
  size="small"
  text="Logo"
>
  <div
    className="container small"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  >
    <Logo
      size="small"
      text="Logo"
    >
      <span
        className="letters small"
      >
        Lo
      </span>
    </Logo>
  </div>
</Component>
`;

exports[`<Thumbnail/> Yellow render should render yellow 1`] = `
<Component
  backgroundColor="yellow"
  size="small"
  text="Logo"
>
  <div
    className="container small"
    style={
      Object {
        "backgroundColor": "yellow",
      }
    }
  >
    <Logo
      size="small"
      text="Logo"
    >
      <span
        className="letters small"
      >
        Lo
      </span>
    </Logo>
  </div>
</Component>
`;
